src/jdk.compiler/share/classes/com/sun/tools/javac/comp/Attr.java

```diff
@@ -1956,7 +1956,7 @@ private boolean isBooleanOrNumeric(Env<AttrContext> env, JCExpression tree) {
         }
         //where
             boolean primitiveOrBoxed(Type t) {
-                return (!t.hasTag(TYPEVAR) && types.unboxedTypeOrType(t).isPrimitive());
+                return (!t.hasTag(TYPEVAR) && !t.hasTag(ERROR) && types.unboxedTypeOrType(t).isPrimitive());
```

Comment:
Good catch - you can also use `!t.isErroneous()` here.

Reply:
Fixed.